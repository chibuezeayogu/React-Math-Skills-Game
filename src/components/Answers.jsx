import React from 'react';
import PropTypes from 'prop-types';

const Answers = ({ selectedNumbers, unselectNumber }) => (
  <div className="col-5">
    {selectedNumbers.map((number, i) => <span 
        key={i}
        role="button"
        tabIndex={0}
        onClick={unselectNumber.bind(null, number)}
      >
        {number}
      </span>
    )
    }
  </div>
);

Answers.propTypes = {
  selectedNumbers: PropTypes.array,
  unselectNumber: PropTypes.func
};

export default Answers;
