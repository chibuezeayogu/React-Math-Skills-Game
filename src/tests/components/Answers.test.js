/* 
global jest 
expect 
*/
import React from 'react';
import { mount } from 'enzyme';
import Answers from '../../components/Answers';

describe('Answers Component', () => {
  const setup = () => {
    const props = {
      selectedNumbers: [1],
      unselectNumber: jest.fn(),
    };

    return mount(<Answers {...props}/>);
  };
  const wrapper = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should unselete a number', () => {
    wrapper.find('span').first().simulate('click');
  });
});
