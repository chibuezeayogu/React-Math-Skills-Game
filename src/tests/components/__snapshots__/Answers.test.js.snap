// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answers Component should render correctly 1`] = `
<Answers
  selectedNumbers={
    Array [
      1,
    ]
  }
  unselectNumber={[MockFunction]}
>
  <div
    className="col-5"
  >
    <span
      key="0"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      1
    </span>
  </div>
</Answers>
`;
